victor/src/layout/dom_traversal.rs

```diff
@@ -39,10 +40,47 @@ pub(super) trait TraversalHandler {
         style: &Arc<ComputedValues>,
         display: DisplayGeneratingBox,
         contents: Contents,
+        box_slot: BoxSlot,
     );
 }
 
-#[allow(unused)]
+pub(super) struct BoxSlot<'dom> {
+    slot: AtomicRefMut<'dom, Option<LayoutBox>>,
+}
+
+impl<'dom> BoxSlot<'dom> {
+    pub fn new(mut slot: AtomicRefMut<'dom, Option<LayoutBox>>) -> Self {
+        *slot = None;
+        Self { slot }
+    }
+
+    pub fn set(mut self, box_: LayoutBox) {
+        *self.slot = Some(box_)
+    }
+}
+
+impl Drop for BoxSlot<'_> {
+    fn drop(&mut self) {
+        assert!(self.slot.is_some(), "failed to set a layout box")
```

Comment:
Won't we want to reset that to `None` for elements which became `display:contents` between two dynamic layouts?

Reply:
I think this code is ok, but your comment lead to find another bug which is fixed in 1340d0679aa4f7e6000ea66a26999af21cc431d7.